packages/flutter/lib/src/widgets/drag_target.dart

```diff
@@ -741,12 +744,14 @@ class _DragTargetState<T extends Object> extends State<DragTarget<T>> {
     setState(() {
       _candidateAvatars.remove(avatar);
     });
-    widget.onAccept?.call(avatar.data! as T);
-    widget.onAcceptWithDetails?.call(DragTargetDetails<T>(data: avatar.data! as T, offset: avatar._lastOffset!));
+    if (avatar.data != null)  {
+      widget.onAccept?.call(avatar.data! as T);
+      widget.onAcceptWithDetails?.call(DragTargetDetails<T>(data: avatar.data! as T, offset: avatar._lastOffset!));
```

Comment:
This should not be included inside the `if` clause since we're planning to make `data` nullable in the newer API.

Reply:
If `data` becomes `nullable` in the new API, then yes it can be moved out of the `if`. But right now, `data!` would crash if `data` is `null`. So it should remain inside the `if` as long as `data` of `onAcceptWithDetails` is not nullable?